service-base/src/main/java/org/eclipse/hono/service/command/Command.java

```diff
@@ -123,7 +123,7 @@ public static Command from(final ProtonDelivery delivery, final Message message,
                         Objects.requireNonNull(tenantId),
                         Objects.requireNonNull(deviceId),
                         correlationId,
-                        replyTo.getResourcePath()[3]);
+                        replyTo.toString().substring(replyTo.getBasePath().length()+1));
```

Comment:
I would rather add a new method (e.g. *asString(int startIdx)*) to `ResourceIdentifier`for this purpose. Or maybe even add something more specific like *getReplyToId()*. WDYT?

Reply:
From the `ResourceIdentifier`s POV I would suggest `getWithoutBasePath()`?

Reply:
Or `getPathWithoutBase()`?